iron-list.html

```diff
@@ -984,11 +1016,21 @@
       }
     },
 
+    /**
+     * Sets the scroll height, that's the height of the content,
+     */
     _updateScrollerSize: function() {
       this._scrollSize = (this._physicalBottom +
           Math.max(this._virtualCount - this._physicalCount - this._virtualStartVal, 0) * this._physicalAverage);
 
-      this.$.items.style.height = this._scrollSize + 'px';
+      var deltaScrollSize = Math.abs(this._scrollSize - this._scrollHeight);
+      var willReachedEnd = this._scrollPosition >= this._scrollSize - this._physicalSize;

```

Comment:
I am not sure if this is enough to know if it reaches the end of the list.  Seems like `this._scrollPosition` is not updated if someone calls `scrollToIndex`.  Can you verify if calling something like `scrollToIndex(10000)` would update `$.items.style.height` correctly?